### PR TITLE
Review 1116 location state region

### PR DESCRIPTION
**Pull request by victordelval**
## Description

Review for PR: https://github.com/regen-network/regen-web/pull/1136

Improvements proposed in the PR review that were not added on time.

See https://github.com/regen-network/regen-web/pull/1136#discussion_r952703758

* Using lazy/suspense with location fields that use the iso3166 dependency
* Also add a generic selector field to use as a fallback in the suspense component.
* Add utility functions to prepare the option list of both selectors for countries and subdivisions

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed
- [ ] once the PR is closed, set up backport PRs for `redwood` and `hambach` (see below)

#### Setting up backport PRs

After merging your PR to `master`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR, `@Mergifyio backport redwood hambach`.

2. If your branch does have merge commits:

    a. Pull latest `master`, `hambach` and `redwood` branches

    b. Create new branches for backports and merge `master` (replace `<PR#>` with your PR #)
    ```
    git checkout -b hambach-backport-<PR#> hambach
    git merge master
    git push origin hambach-backport-<PR#>
    git checkout -b redwood-backport-<PR#> redwood
    git merge master
    git push origin redwood-backport-<PR#>`
    ```

    c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.
  

### How to test

1.

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-components/src/utils/locationStandard.ts [hunk 1]
```diff
@@ -79,3 +81,55 @@ export const getJurisdictionIsoCode = ({
     return stateProvince;
   return `${stateProvince} ${_postalCode}`;
 };
+
+/**
+ *
+ */
```
Comment (flagrede) [inline web-components/src/utils/locationStandard.ts:87]: Was there supposed to be a comment here?

File: web-components/src/utils/locationStandard.ts [hunk 2]
```diff
@@ -79,3 +81,55 @@ export const getJurisdictionIsoCode = ({
...
+  let options = [COUNTRY_OPTION_PLACEHOLDER];
+  if (countryUS) options.push(countryUS);
+  options.push(...otherCountries);
+
+  return options;
+}
+
+/**
+ *
+ */
```
Comment (flagrede) [inline web-components/src/utils/locationStandard.ts:114]: Same

**Review by flagrede — APPROVED** [review]
Looks good !

File: web-components/src/utils/locationStandard.ts (hunk 1 above)
Comment (victordelval) [reply web-components/src/utils/locationStandard.ts]: yes, thanks, i updated it!

**Review by haveanicedavid — APPROVED** [review]
haven’t tested, but code lgtm 👍

**Comment by victordelval** [comment]
@Mergifyio backport redwood hambach

**Pull request merged**